chef/cookbooks/crowbar/recipes/prepare-upgrade-scripts.rb

```diff
@@ -129,7 +129,7 @@
 cinder_volume = roles.include? "cinder-volume"
 neutron = search(:node, "run_list_map:neutron-server").first
 
-if neutron[:neutron][:networking_plugin] == "ml2"
+if !neutron.nil? && neutron[:neutron][:networking_plugin] == "ml2"
```

Comment:
Same as above, if this is causing issues because its executed during the compile phase, maybe we should move it to a function (search for the neutron-server included) and use lazy attribute evaluation so its only executed when its being converged? Maybe a bit too much for this PR but something to consider.